sftp/sftp.tf

```diff
@@ -7,34 +7,77 @@ terraform {
   }
 }
 
+provider "aws" {
+  alias = "ephemeral"
+  region = var.aws_region
+  assume_role {
+    role_arn = var.sftp_account_assume_role
+  }
+}
+
+provider "aws" {
+  alias = "permanent"
+  region = var.aws_region
+  assume_role {
+    role_arn = var.permanent_storage_assume_role
+  }
+}
+
 # #################
 # S3 Storage Bucket
 # #################
 locals {
   s3_bucket_name = lower("${var.transfer_server_s3_bucket_name}-${random_id.aws_s3_bucket_transfer_server.hex}")
+  s3_permanent_bucket = lower("${var.transfer_server_s3_bucket_name}-permanent-${random_id.aws_s3_bucket_transfer_server.hex}")
 }
 
 resource "random_id" "aws_s3_bucket_transfer_server" {
   byte_length = 8
 }
 
-
 resource "aws_s3_bucket" "transfer_server_bucket" {
+  provider      = aws.ephemeral
   bucket        = local.s3_bucket_name
   acl           = "private"
   force_destroy = false
 
+  replication_configuration {
+    role = aws_iam_role.replication.arn
+
+    rules {
+      status = "Enabled"
+
+      destination {
+        bucket        = aws_s3_bucket.permanent_storage.arn
+        storage_class = var.transfer_server_permanent_bucket_storage_class
+      }
+    }
+  }
+
   versioning {
     enabled = true
   }
 
   tags = {
-    Name = local.s3_bucket_name
+    type = "Managed by Terraform"
+  }
+}
+
+resource "aws_s3_bucket" "permanent_storage" {
```

Comment:
is there supposed to be a one of these for this bucket? https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_public_access_block

Reply:
unclear why this comment was resolved as well @pgcloud